docs/misc/config.md

```diff
@@ -0,0 +1,131 @@
+`worldRestart` | The world must be restarted before the config value can be changed.
+
+### ConfigValue
+
+Config values can be built with the provided contexts (if defined) using any of the `#define` methods.
+
+All config value methods take in at least two components:
+* A path representing the name of the variable: a `.` separated string representing the sections the config value is in
+* The default value when no valid configuration is present
+
+The `ConfigValue` specific methods take in two additional components:
+* A validator to make sure the deserialized object is valid
+* A class representing the data type of the config value
+
+```java
+// For some ForgeConfigSpec$Builder builder
+ConfigValue<T> value = builder.comment("Comment")
+  .define("config_value_name", defaultValue);
+```
+
+The values themselves can be obtained using `ConfigValue#get`. The values are additionally cached to prevent multiple readings from files.
+
+#### Additional Config Value Types
+
+* **Range Values**
+  * Description: Value must be between the defined bounds
+  * Class Type: `Comparable<T>`
+  * Method Name: `#defineInRange`
+  * Additional Components:
+    * The minimum and maximum the config value may be
+    * A class representing the data type of the config value
+
+!!! note
+    `DoubleValue`s, `IntValue`s, and `LongValue`s are range values which specify the class as `Double`, `Integer`, and `Long` respectively.
+
+* **Whitelisted Values**
+  * Description: Value must be in supplied collection
+  * Class Type: `T`
+  * Method Name: `#defineInList`
+  * Additional Components:
+    * A collection of the allowed values the configuration can be
+
+* **List Values**
+  * Description: Value is a list of entries
+  * Class Type: `List<T>`
+  * Method Name: `#defineList`, `#defineListAllowEmpty` if list can be empty
+  * Additional Components:
+    * A validator to make sure a deserialized element from the list is valid
+
+* **Enum Values**
+  * Description: An enum value in the supplied collection
+  * Class Type: `Enum<T>`
+  * Method Name: `#defineEnum`
+  * Additional Components:
+    * A getter to convert a string or integer into an enum
+    * A collection of the allowed values the configuration can be
+
+* **Boolean Values**
+  * Description: A `boolean` value
+  * Class Type: `Boolean`
+  * Method Name: `#define`
+
+Registering a Configuration
+---------------------------
+
+Once a `ForgeConfigSpec` has been built, it must be registered to allow Forge to load, track, and sync the configuration settings as required. Configurations should be registered in the mod constructor via `ModLoadingContext#registerConfig`. A configuration can be registered with a given type representing the side the config belongs to, the `ForgeConfigSpec`, and optionally a specific file name for the configuration.
+
+Type   | Loaded | Synced to Client | Stored                         | Default Suffix
+:---:  | :---:  | :---:            | :---:                          | :---
```

Comment:
Something which we may want to look into the cause and solution is the misalignment between the column header and its values. The column values follow the alignment as configured, but the header remains left aligned.

![image](https://user-images.githubusercontent.com/21304337/161408656-b64a164f-14f5-4060-a1e0-6b43c347b213.png)

Reply:
Seems to be a theme issue, so we should probably open an issue for it.

Reply:
Tracked at #431.